cps/templates/edit_book.html

```diff
@@ -37,12 +37,47 @@
       </div>
       <div class="form-group">
         <label for="rating">Rating</label>
-        <input type="text" class="form-control" name="rating" id="rating" value="{% if book.ratings %}{{book.ratings[0].rating}}{% endif %}">
+        <input type="number" min="1" max="5" step="1" class="form-control" name="rating" id="rating" value="{% if book.ratings %}{{book.ratings[0].rating / 2}}{% endif %}">
       </div>
       <div class="form-group">
         <label for="cover_url">Cover URL (jpg)</label>
         <input type="text" class="form-control" name="cover_url" id="cover_url" value="">
       </div>
+      {% if cc|length > 0 %}
+        {% for c in cc %}
+          <div class="form-group">
+            <label for="{{ 'custom_column_' ~ c.id }}">{{ c.name }}</label>
+            {% if c.datatype in ['text', 'series'] %}
```

Comment:
Replace this if block. That fixes the problem with the "Comma-delimited text" type. We also have to take special care of the columns with is_multiple==True in the save routine (treat like tags, split by comma, ...).

```Python
            {% if c.datatype in ['text', 'series'] and not c.is_multiple %}
              <input type="text" class="form-control" name="{{ 'custom_column_' ~ c.id }}" id="{{ 'custom_column_' ~ c.id }}"
              {% if book['custom_column_' ~ c.id]|length > 0 %}
                  value="{{ book['custom_column_' ~ c.id][0].value }}"
              {% endif %}>
            {% endif %}

            {% if c.datatype in ['text', 'series'] and c.is_multiple %}
              <input type="text" class="form-control" name="{{ 'custom_column_' ~ c.id }}" id="{{ 'custom_column_' ~ c.id }}"
              {% if book['custom_column_' ~ c.id]|length > 0 %}
              value="{% for column in book['custom_column_' ~ c.id] %}{{ column.value.strip() }}{% if not loop.last %}, {% endif %}{% endfor %}"{% endif %}>
            {% endif %}

```